### PR TITLE
Account for html entities length

### PR DESCRIPTION
**Pull request by brianburnszd**
### Description

If a string includes a html entity such as a double quote character calculate the encoded length `&quot;`, not the decoded length during truncation.

### Steps to reproduce

```ruby
require 'abbreviato';
string, _truncated = Abbreviato.truncate('""""', max_length: 18)
string.size
# => 25
string
# => "<p>&quot;&quot;&quot;</p>"
```

### CODE REVIEW

**Review by miketineo — APPROVED** [review]
Legendary!

**Review by FLMFreddy — APPROVED** [review]

**Review by swmichaels — APPROVED** [review]
This change is fantastic

**Comment by swmichaels** [comment]
You may want to delete `Gemfile.lock` while you're here and rebuild for Bundler >2. Should bring in any gem bumps as well.

**Comment by barker1977** [comment]
Cc @zendesk/strongbad

File: lib/abbreviato/truncated_sax_document.rb [hunk 1]
```diff
@@ -170,10 +170,20 @@ def max_length_reached?
...
 
-  def truncate_string(string)
+  def truncate_string(decoded_string)
     @truncated = true
     truncate_length = remaining_length - tail.bytesize
-    (string.byteslice(0, truncate_length) || '').scrub('')
+    truncated_string = ''
+    decoded_string.split('').each do |char|
+      encoded_char = @html_coder.encode(char)
+      if encoded_char.bytesize <= truncate_length
```
Comment (alduethadyn) [inline lib/abbreviato/truncated_sax_document.rb]: Maybe:
```
decoded_string.split(‘’).each do |char|
  encoded_char = @html_coder.encode(char)
  break if encoded_char.bytesize > truncated_length

  truncated_string += encoded_char
  truncate_length -= encoded_char.bytesize
end
```

**Review by alduethadyn — APPROVED** [review]
This is awesome! 🎉 

One quick suggestion but looks good to me!

**Review by rcclemente — APPROVED** [review]

**Pull request merged**